src/private/lidar_impl.cpp

```diff
@@ -0,0 +1,191 @@
+                else
+                        ret = ((-122 - value) * 4096) / (5 * 18.18);
+                return ret;
+        }
+
+        static double tilt_volts_to_raw_convert(double value, bool inverse) {
+                double ret;
+                if (inverse)
+                        ret = (value * 5) / 4096;
+                else
+                        ret = (value * 4096) / 5;
+                return ret;
+        }
+    
+        void setApdBias(double bias) {
+                afe_dev->setLongValue(0, adp_bias_volts_to_raw_convert(bias, false), "raw", true);
+        }
+
+        void setTiltVoltage(double voltage) {
+                afe_dev->setLongValue(1, tilt_volts_to_raw_convert(voltage, false), "raw", true);
+        }
+
+        void laserEnable() {
+                pulse_dev->setStringValue(0, "en", "1", true);
+        }
+
+        void laserDisable() {
+                pulse_dev->setStringValue(0, "en", "0", true);
+        }
+
+        void setLaserPulseWidth(long long pulse_width) {
+                pulse_dev->setLongValue(0, pulse_width, "pulse_width_ns", true);
+        }
+
+        void setLaserFrequency(long long frequency) {
+                pulse_dev->setLongValue(0, frequency, "frequency", true);
+        }
+
+        void setChannelSequencerEnableDisable(bool status) {
+                pulse_dev->setBoolValue(status, "sequencer_en");
+        }
+
+        void setChannelSequencerOpMode(const char* mode) {
+                pulse_dev->setStringValue("sequencer_mode", string(mode));
+        }
+
+        void setChannelSequencerOrderAutoMode(const char* order) {
+                pulse_dev->setStringValue("sequencer_auto_cfg", string(order));
+        }
+
+        void setChannelSequencerOrderManualMode(const char* order) {
+                pulse_dev->setStringValue("sequencer_manual_chsel", string(order));
+        }
+
+        void setSequencerPulseDelay(long long ns) {
+                pulse_dev->setLongValue(ns, "sequencer_pulse_delay_ns");
+        }
+
+        void setAllIIoAttributesToDefaultValues() {
+                setChannelSequencerEnableDisable(false);
+                setChannelSequencerOpMode("auto");
+                setChannelSequencerOrderManualMode("0, 0, 0, 0");
+                setChannelSequencerOrderAutoMode("0, 1, 2, 3");
+                setSequencerPulseDelay(248);        
+                laserEnable();
+                setLaserFrequency(50000);
+                setLaserPulseWidth(20);
+                setApdBias(-160);
+                setTiltVoltage(0);
+                // Channel4 must be enabled everytime we want to read any of the other
+                // channels. Enabled it here so that the user doesn't forget to.
+                channelEnableDisable("voltage4", true);
+        }
+
+        ~LidarImpl() {
+                ;
+        }
+    
+private:
+        struct DeviceGeneric* pulse_dev;
+        struct DeviceGeneric* afe_dev;
+        struct DeviceIn* adc_dev;
```

Comment:
Do we need "struct" here?

Reply:
Good catch! 
Removed with the latest commit.